### PR TITLE
Add guides/fullnode

### PR DESCRIPTION
**Pull request by NunoAlexandre**
This PR:
- Changes the format to separate CLI argument setup from the runtime setup, making the guide more readable.
- Added monitoring options and checks
- Reviewed the guide for more clarifying language
- Added more bootnodes for redundancy
- Added demo testnet

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
@gpmayorga would be awesome to have your feedback here when you can. I will run through the guide myself to double check no step is outdated or wrong. I think the node options are now outdated due to to the update to 0.9.43 so I am expecting changes there but hopefully the rest is mostly in place 👍

**Comment by gpmayorga** [comment]
This PR is ready to go !

File: docs/build/guides/how-to-run-a-full-node.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
+---
+id: how-to-run-a-centrifuge-node
+order: 2
+title: How to run a Centrifuge node
+contributors: <Jeroen:jeroen@k-f.co> , <Guillermo:guillermo@k-f.co>
+--- 
+# Running a Centrifuge fullnode
+
```
Comment (hieronx) [inline docs/build/guides/how-to-run-a-full-node.md:8]: It's duplicate in the UI with the page title now
```suggestion
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+- Decide on the [log levels](https://docs.substrate.io/deploy/deployment-options/) depending on your setup
+
+### 1.2 Archive node
+
+Everything same as above but adding `--prune=archive` before the `--` on the CLI arguments
+
+### 1.3 Arguments formatting
+The specific format will depend on how you deploy your node:
+
+Docker/Kuberentes
```
Comment (hieronx) [inline docs/build/guides/how-to-run-a-full-node.md:59]:
```suggestion
Docker/Kubernetes
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    --port=30333 \
+    --rpc-port=9933 \
+    ...
+    -- \
+    ...
+    --sync=fast
+```
+
+### 1.4 Network values
+#### 1.4.1 Mainnet (Centrifuge Polkadot parachain)
```
Comment (hieronx) [inline docs/build/guides/how-to-run-a-full-node.md:80]:
```suggestion
#### 1.4.1 Mainnet (Centrifuge Chain)
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+```
+Chain args:
+```bash
+--chain=centrifuge
+--parachain-id=2031
+--
+--chain=polkadot
+```
+
+#### 1.4.2 Testnet (Centrifuge DEMO)
```
Comment (hieronx) [inline docs/build/guides/how-to-run-a-full-node.md:95]:
```suggestion
#### 1.4.2 Testnet (Centrifuge Demo)
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+# Running a Centrifuge fullnode
+
+## Hardware requirements
+
+- Hardware requirements
+    - minimum: 2+ cores CPU, 4GB+ RAM, 200GB+ free storage space
+    - recommended: 4+ CPU cores, 16GB RAM, 1TB SSD or faster storage
+
+Note: synching and Runtime Upgrades might put extra load on the node, it is recommended to burst the resources until the node is fully synched and have an
+process manager restart the process if it reaches memory limits and hangs or crashes
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:16]: Nit: Language
```suggestion
Note: Synching and Runtime Upgrades might put extra load on the node. It is recommended to burst the resources until the node is fully synched and have a
process manager restart the process if it either reaches memory limits and hangs or crashes
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+for the docker container or in the `node/res/` folder [in the codebase](https://github.com/centrifuge/centrifuge-chain/tree/main/node/res) 
+
+## 2. Recommended deployments
+
+### 2.1 Docker (docker-compose)
+
+You can use the container published on the [Centrifuge Docker Hub repo](https://hub.docker.com/r/centrifugeio/centrifuge-chain)
+or be fully trustless by cloning the [Centrifuge Chain repository](https://github.com/centrifuge/centrifuge-chain/)
+and using the [Dockerfile](https://github.com/centrifuge/centrifuge-chain/blob/main/Dockerfile) (2-4h build time on an average machine),
+if building the image yourself make sure you have checkout the latest tag for the most recent release:
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:119]:
```suggestion
and using the [Dockerfile](https://github.com/centrifuge/centrifuge-chain/blob/main/Dockerfile) (2-4h build time on an average machine).
If building the image yourself, make sure to have check out the latest tag for the most recent release:
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+    - "--port=30333"
+    ...
+    - "--"
+    ...
+    - "--chain=polkadot"
+    - "--sync=fast"
+```
+
+
+**Refer to the CLI arguments on section 1.**
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:159]:
```suggestion
**Refer to the CLI arguments on [section 1.*](/#1-cli-arguments)*
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+title: How to run a Centrifuge node
+contributors: <Jeroen:jeroen@k-f.co> , <Guillermo:guillermo@k-f.co>
+--- 
+# Running a Centrifuge fullnode
+
+## Hardware requirements
+
+- Hardware requirements
+    - minimum: 2+ cores CPU, 4GB+ RAM, 200GB+ free storage space
+    - recommended: 4+ CPU cores, 16GB RAM, 1TB SSD or faster storage
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:13]: Nit: Headline is already _Hardware Requirements_

```suggestion
- Minimum: 2+ cores CPU, 4GB+ RAM, 200GB+ free storage space
- Recommended: 4+ CPU cores, 16GB RAM, 1TB SSD or faster storage
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+    ...
+    -- \
+    ...
+    --sync=fast
+
+[Install]
+WantedBy=multi-user.target
+EOF
+```
+**Refer to the CLI arguments on section 1.**
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:241]:
```suggestion
**Refer to the CLI arguments on [section 1.](/#1-cli-arguments)**
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+
+##### 2. Start the systemd service
+Actually enable the previously generated service and start it.
+
+```bash
+sudo systemctl enable centrifuge.service
+sudo systemctl start centrifuge.service
+```
+
+If everything was set-up correctly, your node should now be starting the process of synchronization.
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:253]:
```suggestion
If everything was set-up correctly, your node should now be start the process of synchronization.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 11]
```diff
@@ -0,0 +1,318 @@
...
+
+```bash
+sudo systemctl status centrifuge.service
+sudo journalctl -u centrifuge.service -f
+```
+
+
+### 3. Test and health monitoring
+
+Once your node is fully synced, you can run a cURL request to see the status of your node, replace localhost for your URL if your node is externally available
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:264]:
```suggestion
Once your node is fully synced, you can run a cURL request to see the status of your node. If your node is externally available, replace `localhost` for your URL.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+curl -H "Content-Type: application/json" \
+-d '{"id":1, "jsonrpc":"2.0", "method": "eth_syncing", "params":[]}' \
+localhost:9933
+```
+
+Expected output if node is synced is `{"jsonrpc":"2.0","result":false,"id":1}`
+
+### 3.1 Use ws-health-exporter
+
+You can monitor your node to make sure it is ready to serve RPC calls using parity's ws-health-exporter.
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:276]:
```suggestion
You can monitor your node to make sure it is ready to serve RPC calls using parity's `ws-health-exporter`.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 13]
```diff
@@ -0,0 +1,318 @@
...
+
+More info [on the parity's Docker Hub page](https://hub.docker.com/r/paritytech/ws-health-exporter)
+
+### 3.2 Monitoring
+As it happens with any blockchain, the storage will run out eventually, it's recommended to monitor your storage or use any kind of auto-scaling storage to account for this.
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:281]:
```suggestion
As it happens with any blockchain, the storage will run out eventually. It is recommended to monitor your storage or use any kind of auto-scaling storage to account for this.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 14]
```diff
@@ -0,0 +1,318 @@
...
+It is also recommended to setup a reverse proxy or an API gateway to monitor the API calls and see the response rate and the response codes to look for errors over time. How to do this is out of the scope of this documentation
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:282]:
```suggestion
It is also recommended to setup a reverse proxy or an API gateway to monitor the API calls and see the response rate and the response codes to look for errors over time. How to do this is out of the scope of this documentation.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 15]
```diff
@@ -0,0 +1,318 @@
...
+localhost:9933
+```
+
+Expected output if node is synced is `{"jsonrpc":"2.0","result":false,"id":1}`
+
+### 3.1 Use ws-health-exporter
+
+You can monitor your node to make sure it is ready to serve RPC calls using parity's ws-health-exporter.
+
+More info [on the parity's Docker Hub page](https://hub.docker.com/r/paritytech/ws-health-exporter)
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:278]:
```suggestion
More info [on the parity's Docker Hub page](https://hub.docker.com/r/paritytech/ws-health-exporter).
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 16]
```diff
@@ -0,0 +1,318 @@
...
+
+Once your node is fully synced, you can run a cURL request to see the status of your node, replace localhost for your URL if your node is externally available
+
+```bash
+curl -H "Content-Type: application/json" \
+-d '{"id":1, "jsonrpc":"2.0", "method": "eth_syncing", "params":[]}' \
+localhost:9933
+```
+
+Expected output if node is synced is `{"jsonrpc":"2.0","result":false,"id":1}`
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:272]:
```suggestion
Expected output if node is synced is `{"jsonrpc":"2.0","result":false,"id":1}`.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 17]
```diff
@@ -0,0 +1,318 @@
...
+### Troubleshooting
+#### Error logs during syncing
+During fast syncing it is expected to see the following error messages on the `[Relaychain]` side.
+
+```bash
+ERROR tokio-runtime-worker sc_service::client::client: [Relaychain] Unable to pin block for finality notification. hash: 0x866f…387c, Error: UnknownBlock: State already discarded [...]
+WARN tokio-runtime-worker parachain::runtime-api: [Relaychain] cannot query the runtime API version: Api called for an unknown Block:  State already discarded [...]
+```
+
+as long as the following logs are seen
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:293]:
```suggestion
As long as the following logs are seen
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 18]
```diff
@@ -0,0 +1,318 @@
...
+
+```bash
+INFO tokio-runtime-worker substrate: [Relaychain] ⚙️  Syncing, target=#18279012 (9 peers), best: #27674 (0x28a4…6fe6), finalized #27648 (0x406d…b89e), ⬇ 1.1MiB/s ⬆ 34.6kiB/s
+INFO tokio-runtime-worker substrate: [Parachain] ⚙️  Syncing 469.4 bps, target=#4306117 (15 peers), best: #33634 (0x79d2…0a45), finalized #0 (0xb3db…9d82), ⬇ 1.3MiB/s ⬆ 2.0kiB/s
+```
+
+everything is working correctly. Once the chain is fully synced the errors logs will go away.
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:300]:
```suggestion
everything is working correctly. Once the chain is fully synced, the errors are expected to vanish.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 19]
```diff
@@ -0,0 +1,318 @@
...
+
+#### Stalled Syncing
+If the chain stops syncing, mostly due to the unavailable blocks then please restart your node. The reason is in most cases that the p2p-view of your node is incorrect at the moment.
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:303]:
```suggestion
If the chain stops syncing, mostly due to the unavailable blocks, then please restart your node. The reason is in most cases that the p2p-view of your node is incorrect at the moment.
```

File: docs/build/guides/how-to-run-a-full-node.md [hunk 20]
```diff
@@ -0,0 +1,318 @@
...
+Resulting in your node dropping the peers and being unable to further sync. A restart helps in theses cases.
+
+Example logs will look like the following:
+```bash
+WARN tokio-runtime-worker sync: [Parachain] 💔 Error importing block 0x88591cb0cb4f66474b189a34abab560e335dc508cb8e7926343d6cf8db6840b7: consensus error: Import failed: Database
+```
+
+#### Changed bootnode or peer identities
+It is common  that bootnode change their p2p-identity leading to the following logs:
```
Comment (wischli) [inline docs/build/guides/how-to-run-a-full-node.md:312]:
```suggestion
It is common that bootnode change their p2p-identity leading to the following logs:
```

**Comment by gpmayorga** [comment]
All suggestions addressed, Ready to go @wischli and @hieronx

**Comment by hieronx** [comment]
Included in https://github.com/centrifuge/documentation/pull/493

**Pull request closed without merging**